src/components/Search/FileUploader.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import { Col, Upload, Button } from 'antd';
+import { UploadOutlined } from '@ant-design/icons';
+
+
+class FileUploader extends Component {
+
+    beforeUpload = file => {
+        const reader = new FileReader();
+        if (this.validateSize(file)) {
+            reader.onload = () => {
+                const json = JSON.parse(reader.result);
+                const geoid = file.name.replace('geo','').replace('.json','');
+
+                if (this.validateFile(json)) {
+                    this.props.onUpload(json, geoid);
+                }
+            };
+            reader.readAsText(file);
+
+            // Prevent upload
+            return false;
+        }
+    }
+
+    validateFile = (json) => {
+        // TODO: validation step
+        console.log(json)
```

Comment:
It's always good to add a log ID. For example: `console.log(`validateFile(): $json)`

Otherwise it will be hard to find in the logs